index/server/main.go

```diff
@@ -300,5 +319,86 @@ func serveDevfileIndex(c *gin.Context) {
 	}()
 
 	// Serve the index.json file
-	c.File(stackIndexPath)
+	iconType := c.Query("icon")
+	if iconType == "base64" {
+		bytes, err := encodeIndexIconToBase64(stackIndexPath)
+		if err != nil {
+			log.Fatal(err.Error())
+		}
+		c.Data(http.StatusOK, http.DetectContentType(bytes), bytes)
+	} else {
+		c.File(stackIndexPath)
+	}
+}
+
+// encodeIndexIconToBase64 encodes all index icons to base64 format given the index file path
+func encodeIndexIconToBase64(indexPath string) ([]byte, error) {
```

Comment:
Also, I'm wondering if rather than calling this function on each request with `icon=base64`, we should be calling this function once and outputting the file to disk, and serving that file when `icon=base64`. What do you think?

Reply:
I can try to add the tests.

Originally I thought we might have more cache files in the registry server given the encoding mechanism performance is not bad, but we need I can create cache files for the encoding.